examples/angular/src/app/guards/can-match-page.guard.ts

```diff
@@ -0,0 +1,42 @@
+import { inject } from '@angular/core';
+import { CanMatchFn, Route, Router, UrlSegment } from '@angular/router';
+
+import { DOTCMS_CLIENT_TOKEN } from '../client-token/dotcms-client';
+import { DotCMSPageAsset } from '@dotcms/angular';
+
+export const canMatchPage: CanMatchFn = async (
+  route: Route,
+  segments: UrlSegment[]
+) => {
+  const router = inject(Router);
+  const client = inject(DOTCMS_CLIENT_TOKEN);
+
+  const path = segments?.map((segment) => segment.path)?.join('/') || '/';
+  const initialUrl = router.getCurrentNavigation()?.initialUrl;
+  const queryParams = initialUrl?.queryParams || {};
+
+  const { mode, variantName, language_id } = queryParams;
+  const personaId = queryParams['com.dotmarketing.persona.id'] || '';
+
+  const pageProps = {
+    path,
+    mode,
+    language_id,
+    variantName,
+    'com.dotmarketing.persona.id': personaId,
+  };
+
+  try {
+    const { entity } = (await client.page.get(pageProps)) as {
```

Comment:
@rjvelazco Why is it necessary to use it as? Shouldn't the lib return the entity with the correct type?

Reply:
@nicobytes The libs are not typed yet. We have a ticket for that. We plan to develop a `@dotcms/types` library to share among the other libraries.

Reply:
`@types/dotcms` actually cc: @nicobytes